simphony/testing/abc_check_engine.py

```diff
@@ -25,6 +30,11 @@ def setUp(self):
         self.items = self.create_dataset_items()
 
     @abc.abstractmethod
+    def container_factory(self, name, mode='w'):
```

Comment:
Right I want to obtain the engine here. Maybe even a best option is engine_factory, for containers I understand ```particles```, ```mesh``` and ```lattice```

The problem with `name` and `mode` is that they are actually used in some tests. For example see test: 

https://github.com/simphony/simphony-common/blob/fix-issue-159/simphony/testing/abc_check_engine.py#L234  

There both variations in the name and opening mode are used. Name dosn't need to be there as we can change it inside the test, but if the opening mode is strictly necessary i can't see how to remove that parameter.

Finally, yes, the idea is to use the ```abc_check_engine``` to test all the functionalities related with the datasets of the H5CUDS class